File: src/main/java/uk/gov/justice/digital/pdf/service/ResourceRepository.java
package uk.gov.justice.digital.pdf.service;

import java.io.IOException;
import java.io.InputStream;
import spark.utils.IOUtils;
import uk.gov.justice.digital.pdf.interfaces.TemplateRepository;

public class ResourceRepository implements TemplateRepository {

    @Override
    public String get(String name) {

        try (InputStream resource = this.getClass().getResourceAsStream(String.format("/templates/%s.html", name))) { // try-with-resources

            return IOUtils.toString(resource);
        }
        catch (IOException ex) {

            return null;
        }
    }
}

File: src/main/kotlin/uk/gov/justice/digital/pdf/service/ResourceRepository.kt
package uk.gov.justice.digital.pdf.service

import spark.utils.IOUtils
import uk.gov.justice.digital.pdf.interfaces.TemplateRepository
import java.io.IOException

class ResourceRepository : TemplateRepository {

    override fun get(name: String?) =

        try { javaClass.getResourceAsStream("/templates/$name.html").use { IOUtils.toString(it) } }
        catch (ex: IOException) { null }
}
